### ISSUE TITLE
[BUG] frac() returns a zero gradient between integer discontinuities

### ISSUE THREAD

**Issue by christophercrouzet** [christophercrouzet/warp#12]
## What

`wp.frac(x)` is `x - trunc(x)`, whose derivative is 1 between integers, but Warp registers it as non-differentiable and its native adjoint is a no-op. Autodiff therefore silently returns zero at smooth inputs.

## Reproduction

```python
import warp as wp

@wp.kernel
def f(x: wp.array[wp.float32], y: wp.array[wp.float32]):
    y[0] = wp.frac(x[0])

for device in ("cpu", "cuda:0"):
    x = wp.array([1.25], dtype=wp.float32, requires_grad=True, device=device)
    y = wp.zeros(1, dtype=wp.float32, requires_grad=True, device=device)
    with wp.Tape() as tape:
        wp.launch(f, dim=1, inputs=[x], outputs=[y], device=device)
    tape.backward(y)
    print(device, y.numpy(), x.grad.numpy())
```

Both devices print value `0.25` and gradient `0.0`; centered finite differences give `1.0`.

## Why / when it matters

Losses using a fractional phase, sawtooth coordinate, or periodic indexing term receive no gradient even when evaluated away from an integer discontinuity.

## Environment

Reproduced against NVIDIA/warp `main` at `b1922eba85bd32af5e393c829c0b22e70ea7abea` (Warp 1.18.0.dev5) on Linux.

Reproduced on CPU and CUDA.
